TweetMate/src/main/java/com/ybsystem/tweetmate/activities/ActivityBase.java

```diff
@@ -0,0 +1,131 @@
+package com.ybsystem.tweetmate.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.MenuItem;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.ybsystem.tweetmate.R;
+import com.ybsystem.tweetmate.application.TweetMateApp;
+import com.ybsystem.tweetmate.resources.ResColor;
+import com.ybsystem.tweetmate.databases.PrefTheme;
+
+public abstract class ActivityBase extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        TweetMateApp.setActivity(this);
+
+        // Set
+        setTheme();
+        setBackButton();
+
+        // Init resource
+        ResColor.init();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        TweetMateApp.setActivity(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        switch (TweetMateApp.getActivityName(getLocalClassName())) {
+            case "MainActivity":
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "MainActivity" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1192)